### PR TITLE
Update E2E test version references

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/test/E2ETests.csproj [hunk 1]
```diff
@@ -42,13 +42,10 @@
...
   <ItemGroup Condition=" '$(AZURE_IOT_LOCALPACKAGES)' != '' ">
-    <PackageReference Include="Microsoft.Azure.Devices" Version="1.*" />
-    <PackageReference Include="Microsoft.Azure.Devices.Authentication" Version="1.*" />
-    <PackageReference Include="Microsoft.Azure.Devices.Client" Version="1.*" />
-    <PackageReference Include="Microsoft.Azure.Devices.Provisioning.Client" Version="1.*" />
-    <PackageReference Include="Microsoft.Azure.Devices.Provisioning.Transport.Amqp" Version="1.*" />
-    <PackageReference Include="Microsoft.Azure.Devices.Provisioning.Transport.Http" Version="1.*" />
-    <PackageReference Include="Microsoft.Azure.Devices.Provisioning.Transport.Mqtt" Version="1.*" />
-    <PackageReference Include="Microsoft.Azure.Devices.Provisioning.Service" Version="1.*" />
+    <PackageReference Include="Microsoft.Azure.Devices" Version="2.*" />
```
Comment (drwill-ms) [inline e2e/test/E2ETests.csproj:45]: trim and v2

File: vsts/build-test-nuget.yml [hunk 2]
```diff
@@ -62,8 +62,6 @@ jobs:
           FRAMEWORK: netcoreapp2.1.30
         .Net Framwork 4.7.2:
           FRAMEWORK: net472
-        .Net Framework 4.5.1:
```
Comment (drwill-ms) [inline vsts/build-test-nuget.yml:65]: we don't support/test 4.5.1 any more

File: vsts/releaseTest.ps1 [hunk 3]
```diff
@@ -2,14 +2,14 @@
 # Licensed under the MIT license. See LICENSE file in the project root for full license information.
 Function IsWindows() 
 {
-	return ([Environment]::OSVersion.Platform -eq [System.PlatformID]::Win32NT)
```
Comment (drwill-ms) [inline vsts/releaseTest.ps1:5]: tabs to spaces

File: vsts/releaseTest.ps1 [hunk 4]
```diff
@@ -26,20 +26,20 @@ Invoke-Expression $runTestCmd
...
+    exit 1
 }
-else 
+else
 {
-	Write-Host "Testing was successful!"
-	exit 0
+    Write-Host "Testing was successful!"
+    exit 0
 }
```
Comment (brycewang-microsoft) [inline vsts/releaseTest.ps1:45]: Do we need a new line at end of this file?

**Review by brycewang-microsoft — APPROVED** [review]

File: vsts/releaseTest.ps1 (hunk 4 above)
Comment (drwill-ms) [reply vsts/releaseTest.ps1:45]: Need? No, but we generally like to have one.

**Pull request merged**